### PR TITLE
circulation infos refactoring 

### PR DESCRIPTION
**Pull request by zannkukai**
The message return by the backend when a user is blocked has changed.
This commit adapts the UI to this change and display the error message
returned by the backend directly.
This commit also adds counter on tabs of the patron circulation detail
page. Available counter are for checkin/checkout, pickup and pending
tabs.

Small correction on the template detail view.

Closes rero/rero-ils#1278
Closes rero/rero-ils#1281

Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Dependencies

 * ng-core 0.13 (run `npm ci` the new version of ng-core should be installed)
 * https://github.com/rero/rero-ils/tree/US1643-borrow-limits

## How to test?

image

* Use the professional interface to block a user. 
* Go in the circulation interface for this patron ; you should see this message in red (see screenshot)
* Unblock the patron
* Define some (small) limits on a patron_type and assign this patron_type to this patron
* Do some checkout operation until the define limit is reached.
* Reload the circulation interface for this patron : you should see a message in red.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

**Comment by Garfield-fr** [comment]
I consider the message system to be too large. Soon you don't see the important part.

File: projects/admin/src/app/circulation/patron/card/card.component.html [hunk 1]
```diff
@@ -38,32 +38,36 @@ <h3>
...
-          <strong class="alert-heading">{{ note.type | translate }}</strong>
-          <p class="mb-0" [innerHTML]="note.content"></p>
-        </ng-container>
-      </div>
+  </div>
 
+  <!-- NOTES-->
+  <div *ngIf="patron.notes" class="alert alert-warning" role="alert">
+    <ng-container *ngFor="let note of patron.notes">
+      <strong class="alert-heading">{{ note.type.toString() | translate }}</strong>
```
Comment (Garfield-fr) [inline projects/admin/src/app/circulation/patron/card/card.component.html:59]: `<strong class="alert-heading" translate>{{ note.type.toString() }}</strong>`

File: projects/admin/src/app/record/detail-view/template-detail-view/template-detail-view.component.html [hunk 2]
```diff
@@ -30,7 +30,7 @@ <h2 id="template-name" class="pl-0" *ngIf="record.metadata.name">
           </span>
         </dt>
         <dd id="template-description" [ngClass]="ddCssClass">
-          {{ record.metadata.description }}
+          {{ record.metadata.description | nl2br }}
         </dd>
```
Comment (Garfield-fr) [inline projects/admin/src/app/record/detail-view/template-detail-view/template-detail-view.component.html:34]: `<dd id="template-description" [ngClass]="ddCssClass" [innerHtml]="record.metadata.description | nl2br">`

File: projects/admin/src/app/service/patron.service.ts [hunk 3]
```diff
@@ -168,6 +168,16 @@ export class PatronService {
     return this.getLoans(query, '-end_date');
   }
 
+  /**
+   * Get circulation statistics about a patron
+   * @param patronPid - string : the patron pid to search
+   * @return Observable
+   */
+  getCirculationInformations(patronPid: string): Observable<any> {
+    const url = `/api/patrons/${patronPid}/circulation_informations`;
```
Comment (Garfield-fr) [inline projects/admin/src/app/service/patron.service.ts:177]: Inject `ApiService` to `PatronService` and use `[this._apiService.getEndpointByType('patrons'), patronPid, 'circulation_informations'].join('/')`

File: projects/admin/src/app/circulation/patron/card/card.component.html [hunk 4]
```diff
@@ -26,7 +26,7 @@
           <h3>
             <a [routerLink]="['/records', 'patrons','detail', patron.pid]">
               <span id="patron-last-name" *ngIf="patron.last_name">{{ patron.last_name }}</span>
-              <span id="patron-first-name" *ngIf="patron.first_name">, {{ patron.first_name }} </span>
+              <span id="patron-first-name" *ngIf="patron.first_name">, {{ patron.first_name }}</span>
             </a>
```
[the remark this answers is not in the document]
Comment (zannkukai) [reply projects/admin/src/app/circulation/patron/card/card.component.html:30]: at UCL, we have some students (mainly from Asia or Africa) without any firstname.
Asked the question to PO to found a solution

File: projects/admin/src/app/circulation/patron/card/card.component.ts [hunk 5]
```diff
@@ -15,56 +15,28 @@
...
-    }
-    return this.patron.notes.map((note: any) => {
-      return {
-        type: note.type,
-        content: this._sanitizer.bypassSecurityTrustHtml(
-          note.content.replace('\n', '<br>'))
-      };
-    });
+
+  getBoootstrapColor(level: string): string {
```
Comment (AoNoOokami) [inline projects/admin/src/app/circulation/patron/card/card.component.ts:38]: There are 3 `o` in the method definition

**Review by AoNoOokami — CHANGES_REQUESTED** [review]
When I create a patron type to add limits, clicking on one of the limit toggle switch activates `Yearly subscription`. I couldn't test further because of that.

limits

**Comment by AoNoOokami** [comment]
> When I create a patron type to add limits, clicking on one of the limit toggle switch activates `Yearly subscription`. I couldn't test further because of that.
> 
> limits

Due to the wrong version of ng-core.

**Review by AoNoOokami — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

**Pull request merged**